Infinite recursion in NewArrayBufferWithContents

```
warning: mozjs_sys@0.128.3-6arg: src/jsapi.cpp: In function ‘JSObject* glue::NewArrayBufferWithContents(JSContext*, size_t, void*)’:
warning: mozjs_sys@0.128.3-6arg: src/jsapi.cpp:158:11: warning: infinite recursion detected [-Winfinite-recursion]
warning: mozjs_sys@0.128.3-6arg:   158 | JSObject* NewArrayBufferWithContents(JSContext* cx, size_t nbytes,
warning: mozjs_sys@0.128.3-6arg:       |           ^~~~~~~~~~~~~~~~~~~~~~~~~~
warning: mozjs_sys@0.128.3-6arg: src/jsapi.cpp:161:36: note: recursive call
warning: mozjs_sys@0.128.3-6arg:   161 |   return NewArrayBufferWithContents(cx, nbytes, contents);
warning: mozjs_sys@0.128.3-6arg:       |          ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
```

https://github.com/servo/mozjs/blob/18ab014e77eee726545ed544ea83d555bdfa46a3/mozjs-sys/src/jsapi.cpp#L158-L161

--- a/mozjs-sys/src/jsapi.cpp
+++ b/mozjs-sys/src/jsapi.cpp
@@ -158,7 +158,7 @@ JSObject* NewExternalArrayBuffer(JSContext* cx, size_t nbytes, void* contents,
 JSObject* NewArrayBufferWithContents(JSContext* cx, size_t nbytes,
                                      void* contents) {
   js::UniquePtr<void, JS::FreePolicy> dataPtr{contents};
-  return NewArrayBufferWithContents(cx, nbytes, contents);
+  return JS::NewArrayBufferWithContents(cx, nbytes, std::move(dataPtr));
 }
 
 // Reexport some methods

